src/reducers/fileTree.spec.tsx

```diff
@@ -38,20 +43,78 @@ describe(__filename, () => {
   describe('reducer', () => {
     it('builds and loads a tree', () => {
       const version = createInternalVersion(fakeVersion);
-      const state = reducer(undefined, actions.buildTree({ version }));
+      const comparedToVersionId = nextUniqueId();
+      const state = reducer(
+        undefined,
+        actions.buildTree({ comparedToVersionId, version }),
+      );
 
       expect(state).toEqual({
         ...initialState,
+        comparedToVersionId,
         forVersionId: version.id,
         tree: buildFileTree(version),
       });
     });
+
+    it('invalidates the tree when reloading a version', () => {
+      let state = reducer(
+        undefined,
+        actions.buildTree({
+          comparedToVersionId: null,
+          version: createInternalVersion(fakeVersion),
+        }),
+      );
+      state = reducer(
+        state,
+        versionsActions.loadVersionInfo({ version: fakeVersion }),
+      );
+
+      expect(state).toEqual(
+        expect.objectContaining({
+          forVersionId: undefined,
+        }),
+      );
+    });
+
+    it('does not reset the tree when reloading another version', () => {
+      const oldVerisonId = nextUniqueId();
+      const newVersionId = nextUniqueId();
+      let state = reducer(
+        undefined,
+        actions.buildTree({
+          comparedToVersionId: null,
+          version: createInternalVersion({ ...fakeVersion, id: oldVerisonId }),
+        }),
+      );
+      state = reducer(
+        state,
+        versionsActions.loadVersionInfo({
+          version: { ...fakeVersion, id: newVersionId },
+        }),
+      );
+      state = reducer(
+        state,
+        versionsActions.loadVersionInfo({
+          version: { ...fakeVersion, id: newVersionId },
```

Comment:
Why does this dispatch the same version twice? Is it just to be extra sure that the reducer handles multiple dipstaches? If so, I suggest adding a comment explaining that because otherwise the duplicate action looks like it might have been a mistake.

Reply:
My original thought is to *reloading* another version. After reconsidering it, I think we do not need it, so I removed the second dispatch (just to test the case of *loading* another version).

I pushed the updates.